src/variable.jl

```diff
@@ -127,8 +126,6 @@ end
 fix!(x::Variable, v::Number) = fix!(x, fill(v, (1, 1)))
 
 function free!(x::Variable)
-    # TODO this won't work if :fixed appears other than at the end of x.sets
-    x.sets[end] == :fixed && pop!(x.sets)
     x.vexity = AffineVexity()
```

Comment:
This is mostly unrelated to the PR, but I noticed something: isn't it possible that `x` could have been something other than affine before fixing it as a constant? If that can occur, in such cases `x` would not actually have its vexity reset by freeing but rather changed entirely.

Reply:
I see what you mean, but I think `Variable`'s only have `ConstVexity()` or `AffineVexity()`, where `ConstVexity()` corresponds to the fixed case. I think other vexities don't quite make sense for a `Variable` (but instead an expression involving a variable might have a different vexity, e.g. `log(x)` would be concave). To double check this, I searched the source for `.vexity`, since that would be the way to set a `Variable` to another vexity (since also searching for `field` and `property` doesn't show anything relevant), and only saw `.vexity` show up in a getter, in `free!`, in a show method, and in the line I changed,
```julia
        if x.vexity == ConstVexity()
            # do exactly what we would for a constant
            ...
```
so I think indeed, Variable's vexities are not set to anything but `AffineVexity()` (and `ConstVexity()` in `fix!`).

Reply:
Just to say, I realized that actually I probably should have written `vexity(x)` instead of `x.vexity` in that if condition (we only need direct access for setting it, not accessing), so I pushed one (hopefully last) commit to change that.